src/operators.jl

```diff
@@ -236,175 +232,21 @@ julia> collect(edges(f))
-            for v in neighbors(h, u)
-                add_edge!(r, u, v)
-            end
-        end
-    end
-    return r
-end
-
-@traitfn function symmetric_difference(g::AG, h::AG) where {AG; !IsDirected{AG}}
-    limit = min(nv(g), nv(h))
-    r = SimpleDiGraph(max(nv(g), nv(h)))
-    for u in 1:limit
-        gnv = length(neighbors(g, u))
-        hnv = length(neighbors(h, u))
-        l1 = outneighbors(g, u)
-        l2 = outneighbors(h, u)
-        ptr1 = searchsortedfirst(l1, u)
-        ptr2 = searchsortedfirst(l2, u)
-        while ptr1 <= gnv && ptr2 <= hnv
-            if l1[ptr1] < l2[ptr2]
-                while ptr1 <= gnv && l1[ptr1] < l2[ptr2]
-                    add_edge!(r, u, l1[ptr1])
-                    ptr1 += 1
-                end
-            else
-                while ptr2 <= hnv && l1[ptr1] > l2[ptr2]
-                    add_edge!(r, u, l2[ptr2])
-                    ptr2 += 1
-                end
-            end
-            if ptr1 <= gnv && ptr2 <= hnv && l1[ptr1] == l2[ptr2]
-                ptr1 += 1
-                ptr2 += 1
-            end
-        end
-        while ptr2 <= hnv
-            add_edge!(r, u, l2[ptr2])
-            ptr2 += 1
-        end
-        while ptr1 <= gnv
-            add_edge!(r, u, l1[ptr1])
-            ptr1 += 1
-        end
-    end
-    if limit < nv(g)
-        for u in limit+1:nv(g)
-            l1 = neighbors(g, u)
-            i = searchsortedfirst(l1, u)
-            while i <= length(l1)
-                add_edge!(r, u, l1[i])
-                i += 1
-            end
-        end
-    end
-    if limit < nv(h)
-        for u in limit+1:nv(h)
-            l1 = neighbors(h, u)
-            i = searchsortedfirst(l1, u)
-            while i <= length(l1)
-                add_edge!(r, u, l1[i])
-                i += 1
-            end
-        end
-    end
-    return r
-end
-
-
-"""
-    union(g, h)
-
-Return a graph that combines graphs `g` and `h` by taking the set union
-of all vertices and edges.
-
-### Implementation Notes
-Preserves the eltype of the input graph. Will error if the
-number of vertices in the generated graph exceeds the eltype.
-
-# Examples
-```jldoctest
-julia> using LightGraphs
-
-julia> g = SimpleGraph(3); h = SimpleGraph(5);
-
-julia> add_edge!(g, 1, 2);
-
-julia> add_edge!(g, 1, 3);
-
-julia> add_edge!(h, 3, 4);
-
-julia> add_edge!(h, 3, 5);
-
-julia> add_edge!(h, 4, 5);
-
-julia> f = union(g, h);
-
-julia> collect(edges(f))
-5-element Array{LightGraphs.SimpleGraphs.SimpleEdge{Int64},1}:
- Edge 1 => 2
- Edge 1 => 3
- Edge 3 => 4
- Edge 3 => 5
- Edge 4 => 5
-```
-"""
-function union(g::T, h::T) where T <: AbstractSimpleGraph
-    gnv = nv(g)
-    hnv = nv(h)
-
-    r = T(max(gnv, hnv))
-    r.ne = ne(g)
-    for i in vertices(g)
-        r.fadjlist[i] = deepcopy(g.fadjlist[i])
-        if is_directed(g)
-            r.badjlist[i] = deepcopy(g.badjlist[i])
-        end
+function symmetric_difference(g::AG, h::AG) where {AG<:AbstractGraph}
```

Comment:
What would be the expected behavior of  this method for weighted edges? I think we need to be a bit more clear with this, but I would suggest we do that in a follow up pr.

Reply:
It depends on how one defines equality of weighted edges. Right now we assume two edges are equal if their sources and dests are equal. Other graph types are free to change that (though things might break downstream; I haven't really tested that).